src/ray/object_manager/object_manager.cc

```diff
@@ -0,0 +1,460 @@
+  SchedulePull(object_id, config_.pull_timeout_ms);
+};
+
+ray::Status ObjectManager::Pull(const ObjectID &object_id,
+                                const ClientID &client_id) {
+  Status status = GetMsgConnection(
+      client_id,
+      [this, object_id](SenderConnection::pointer client){
+        Status status = ExecutePull(object_id, client);
+      });
+  return status;
+};
+
+ray::Status ObjectManager::ExecutePull(const ObjectID &object_id,
+                                       SenderConnection::pointer conn) {
+  size_t msg_type = OMMessageType_PullRequest;
+  boost::system::error_code ec;
+  boost::asio::write(conn->GetSocket(),
+                     boost::asio::buffer(&msg_type,
+                                         sizeof(msg_type)),
+                     ec);
+  flatbuffers::FlatBufferBuilder fbb;
+  auto message = CreatePullRequest(fbb,
+                                   fbb.CreateString(client_id_.binary()),
+                                   fbb.CreateString(object_id.binary()));
+  fbb.Finish(message);
+  size_t length = fbb.GetSize();
+  std::vector<boost::asio::const_buffer> buffer;
+  buffer.push_back(boost::asio::buffer(&length, sizeof(length)));
+  buffer.push_back(boost::asio::buffer(fbb.GetBufferPointer(), length));
+  boost::asio::write(conn->GetSocket(), buffer);
+  return ray::Status::OK();
+};
+
+ray::Status ObjectManager::Push(const ObjectID &object_id,
+                                const ClientID &client_id) {
+  ray::Status status;
+  status = GetTransferConnection(
+      client_id,
+      [this, object_id](SenderConnection::pointer conn){
+        ray::Status status = QueuePush(object_id, conn);
+      });
+  return status;
+};
+
+ray::Status ObjectManager::Cancel(const ObjectID &object_id) {
+  // TODO(hme): Account for pull timers.
+  ray::Status status = object_directory_->Cancel(object_id);
+  return ray::Status::OK();
+};
+
+ray::Status ObjectManager::Wait(const vector<ObjectID> &object_ids,
+                                uint64_t timeout_ms,
+                                int num_ready_objects,
+                                const WaitCallback &callback) {
+  // TODO: Implement wait.
+  return ray::Status::OK();
+};
+
+ray::Status ObjectManager::GetMsgConnection(
+    const ClientID &client_id,
+    std::function<void(SenderConnection::pointer)> callback){
+
+  ray::Status status = Status::OK();
+  if(message_send_connections_.count(client_id) > 0){
+    callback(message_send_connections_[client_id]);
+  } else {
+    status = object_directory_->GetInformation(
+        client_id,
+        [this, callback](RemoteConnectionInfo info){
+          Status status = CreateMsgConnection(info, callback);
+        },
+        [this](const Status &status){
+          // TODO: deal with failure.
+        });
+  }
+  return status;
+};
+
+ray::Status ObjectManager::CreateMsgConnection(const RemoteConnectionInfo &info,
```

Comment:
Msg -> Message

Reply:
Okay, I am making the msg -> message changes, but I think the google style doc says widely accepted conventions are okay, and I think msg qualifies.